src/functions-and-arrays.js

```diff
@@ -1,41 +1,145 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(number1, number2) {
+  if (number1 > number2) {
+    return number1;
+  }
+  else if (number1 < number2) {
+    return number2;
+  }
+  else {
+    return number1;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(arr) {
+  let longestWord = ""
+  if (!arr.length) {
+    return null;
+  }
+  else if (arr.length === 1) {
+    return arr[0];
+  }
+  for (i=0; i < arr.length; i++) {
+    if (arr[i].length > longestWord.length) {
+      longestWord = arr[i];
+    } 
+  }
+  return longestWord;
+}
 
+console.log(findLongestWord(words))
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(arr) {
+  let sum = 0;
+  if (!arr.length) {
+    return 0;
+  }
+  else if (arr.length>0) {
+    for (i=0; i<arr.length; i++) {
+      sum += arr[i];
+    }
+    return sum
+  }
+}
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
-
-
+function sum(arr) {
+  let sum = 0;
+  for(let i=0; i<arr.length; i++) {
+    if (typeof arr[i] === "number") {
+      sum += arr[i];
+    }
+    else if (typeof arr[i] === "string") {
+      sum += arr[i].length;
+    }
+    else if (typeof arr[i] === "boolean") {
+      if (arr[i] === true) {
+        sum += 1;
+      }
```

Comment:
Also `true` and `false` equals to `1` and `0`, so

`else if (typeof arr[i] === "boolean") {
      sum += arr[i];
    }`